File: C++/src/Solution901_1000/Solution917.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution917.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

string reverseOnlyLetters(string s) {
  int left = 0, right = s.size() - 1;
  while (left < right){
    while (left < right && !isalpha(s[left])) ++left;
    while (left < right && !isalpha(s[right])) --right;
    swap(s[left++], s[right--]);
  }
  return s;
}

File: Java/src/main/java/QuestionBank/Solution901_1000/Solution917.java
package QuestionBank.Solution901_1000;

/**
 * @author windows
 */
public class Solution917 {
    public String reverseOnlyLetters(String s) {
        char[] chars = s.toCharArray();
        int left = 0;
        int right = chars.length - 1;
        while(left < right){
            while(left < right && !Character.isLetter(chars[left])){
                left++;
            }
            while(left < right && !Character.isLetter(chars[right])){
                right--;
            }
            if(left < right){
                char temp = chars[left];
                chars[left] = chars[right];
                chars[right] = temp;
                left++;
                right--;
            }
        }
        return new String(chars);
    }


    public static void main(String[] args) {
        Solution917 solution917 = new Solution917();
    }
}